flink-connector-aws/flink-connector-dynamodb/src/test/java/org/apache/flink/connector/dynamodb/source/enumerator/DynamoDbStreamsSourceEnumeratorStateSerializerTest.java

```diff
@@ -0,0 +1,173 @@
+        Set<DynamoDbStreamsShardSplit> unassignedSplits =
+                Stream.of(getTestSplit(generateShardId(1)), getTestSplit(generateShardId(2)))
+                        .collect(Collectors.toSet());
+        String lastSeenShardId = "shardId-000000000002";
+        DynamoDbStreamsSourceEnumeratorState initialState =
+                new DynamoDbStreamsSourceEnumeratorState(unassignedSplits, lastSeenShardId);
+
+        DynamoDbStreamsShardSplitSerializer splitSerializer =
+                new DynamoDbStreamsShardSplitSerializer();
+        DynamoDbStreamsSourceEnumeratorStateSerializer serializer =
+                new DynamoDbStreamsSourceEnumeratorStateSerializer(splitSerializer);
+        DynamoDbStreamsSourceEnumeratorStateSerializer wrongVersionStateSerializer =
+                new WrongVersionStateSerializer(splitSerializer);
+
+        byte[] serialized = wrongVersionStateSerializer.serialize(initialState);
+
+        assertThatExceptionOfType(VersionMismatchException.class)
+                .isThrownBy(
+                        () ->
+                                serializer.deserialize(
+                                        wrongVersionStateSerializer.getVersion(), serialized))
+                .withMessageContaining(
+                        "Trying to deserialize DynamoDbStreamsSourceEnumeratorState serialized with unsupported version")
+                .withMessageContaining(String.valueOf(serializer.getVersion()))
+                .withMessageContaining(String.valueOf(wrongVersionStateSerializer.getVersion()));
+    }
+
+    @Test
+    void testDeserializeWithWrongVersionSplitSerializer() throws Exception {
+        Set<String> completedShardIds =
+                Stream.of("shardId-000000000001", "shardId-000000000002", "shardId-000000000003")
+                        .collect(Collectors.toSet());
+        Set<DynamoDbStreamsShardSplit> unassignedSplits =
+                Stream.of(getTestSplit(generateShardId(1)), getTestSplit(generateShardId(2)))
+                        .collect(Collectors.toSet());
+        String lastSeenShardId = "shardId-000000000002";
+        DynamoDbStreamsSourceEnumeratorState initialState =
+                new DynamoDbStreamsSourceEnumeratorState(unassignedSplits, lastSeenShardId);
+
+        DynamoDbStreamsShardSplitSerializer splitSerializer =
+                new DynamoDbStreamsShardSplitSerializer();
+        DynamoDbStreamsSourceEnumeratorStateSerializer serializer =
+                new DynamoDbStreamsSourceEnumeratorStateSerializer(splitSerializer);
+        DynamoDbStreamsShardSplitSerializer wrongVersionSplitSerializer =
+                new WrongVersionSplitSerializer();
+        DynamoDbStreamsSourceEnumeratorStateSerializer wrongVersionStateSerializer =
+                new DynamoDbStreamsSourceEnumeratorStateSerializer(wrongVersionSplitSerializer);
+
+        byte[] serialized = wrongVersionStateSerializer.serialize(initialState);
```

Comment:
nit: this should be the correct version, wrong version should be below

Reply:
Copied this from kinesis: https://github.com/apache/flink-connector-aws/blob/main/flink-connector-aws/flink-connector-aws-kinesis-streams/src/test/java/org/apache/flink/connector/kinesis/source/enumerator/KinesisStreamsSourceEnumeratorStateSerializerTest.java#L64

Reply:
updated